packages/web-frontend/src/containers/UserMenu/index.js

```diff
@@ -213,58 +140,38 @@ const UserMenu = ({
 
   const showRegisterButton = !isUserLoggedIn && isCustomLandingPage;
 
+  const MenuComponent = isDrawer ? DrawerMenu : PopoverMenu;
+  const menuItems = isCustomLandingPage ? customLandingPageMenuItems : baseMenuItems;
+
   return (
     <UserMenuContainer secondaryColor={secondaryColor}>
-      {isUserLoggedIn ? (
-        <UsernameContainer isCustomLandingPage={isCustomLandingPage}>
-          {currentUserUsername}
-        </UsernameContainer>
-      ) : (
-        <>
-          {showRegisterButton && (
-            <RegisterButton onClick={onClickRegister}>Register</RegisterButton>
-          )}
-          <SignInButton onClick={onClickSignIn} secondaryColor={secondaryColor}>
-            {signInText}
-          </SignInButton>
-        </>
+      {/** Only display the user info before the menu button if user is not on a mobile device, as the drawer menu handles this for mobile devices */}
+      {!isDrawer && (
+        <UserInfo
+          showRegisterButton={showRegisterButton}
+          currentUserUsername={currentUserUsername}
+          isCustomLandingPage={isCustomLandingPage}
+          onClickRegister={onClickRegister}
+          onClickSignIn={onClickSignIn}
+          signInText={signInText}
+          secondaryColor={secondaryColor}
+        />
       )}
-      <div>
-        <StyledMenuButton onClick={toggleUserMenu} disableRipple id="user-menu-button">
-          <StyledMenuIcon />
-        </StyledMenuButton>
-        <Popover
-          PaperProps={{ style: { backgroundColor: primaryColor } }}
-          open={menuOpen}
-          anchorEl={() => document.getElementById('user-menu-button')}
-          onClose={closeUserMenu}
-          anchorOrigin={{
-            vertical: 'bottom',
-            horizontal: 'right',
-          }}
-          transformOrigin={{
-            vertical: 'top',
-            horizontal: 'right',
-          }}
-        >
-          <MenuList>
-            {menuItems.map(({ action, text, type, url }) => (
-              <MenuListItem key={text}>
-                <MenuItem
-                  type={type}
-                  onClick={() => {
-                    if (action) action();
-                    closeUserMenu();
-                  }}
-                  href={url}
-                >
-                  {text}
-                </MenuItem>
-              </MenuListItem>
-            ))}
-          </MenuList>
-        </Popover>
-      </div>
+      <StyledMenuButton onClick={toggleUserMenu} disableRipple id="user-menu-button">
+        <StyledMenuIcon />
+      </StyledMenuButton>
+      <MenuComponent
+        menuOpen={menuOpen}
+        primaryColor={primaryColor}
+        onCloseMenu={closeUserMenu}
+        secondaryColor={secondaryColor}
+        onClickSignIn={onClickSignIn}
+        onClickRegister={onClickRegister}
+        signInText={signInText}
+        menuItems={menuItems}
+        isUserLoggedIn={isUserLoggedIn}
+        currentUserUsername={currentUserUsername}
+      />
```

Comment:
```suggestion
      <MenuComponent
        menuOpen={menuOpen}
        primaryColor={primaryColor}
        onCloseMenu={closeUserMenu}
      >
      <MenuItem onClick={onAttemptUserLogout}>Logout<MenuItem/>
      ...
      </MenuComponent>
```
The component component would look something like this and the MenuItem would be a component from the MenuList file.

Reply:
Also it looks like most of the props passed to MenuComponent aren't needed anymore.